Src/Newtonsoft.Json/JsonWriter.Async.cs

```diff
@@ -0,0 +1,1573 @@
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return cancellationToken.CancelledAsync();
+
+            WriteValue(value);
+            return AsyncUtils.CompletedTask;
+        }
+
+        /// <summary>
+        /// Asynchronously writes a <see cref="Nullable{T}"/> of <see cref="sbyte"/> value.
+        /// </summary>
+        /// <param name="value">The <see cref="Nullable{T}"/> of <see cref="sbyte"/> value to write.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>A <see cref="Task"/> that represents the asynchronous operation.</returns>
+        /// <remarks>The default behaviour is to execute synchronously, returning an already-completed task. Derived
+        /// classes can override this behaviour for true asychronousity.</remarks>
+        [CLSCompliant(false)]
+        public virtual Task WriteValueAsync(sbyte? value, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return cancellationToken.CancelledAsync();
+
+            WriteValue(value);
+            return AsyncUtils.CompletedTask;
+        }
+
+        /// <summary>
+        /// Asynchronously writes a <see cref="short"/> value.
+        /// </summary>
+        /// <param name="value">The <see cref="short"/> value to write.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>A <see cref="Task"/> that represents the asynchronous operation.</returns>
+        /// <remarks>The default behaviour is to execute synchronously, returning an already-completed task. Derived
+        /// classes can override this behaviour for true asychronousity.</remarks>
+        public virtual Task WriteValueAsync(short value, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return cancellationToken.CancelledAsync();
+
+            WriteValue(value);
+            return AsyncUtils.CompletedTask;
+        }
+
+        /// <summary>
+        /// Asynchronously writes a <see cref="Nullable{T}"/> of <see cref="short"/> value.
+        /// </summary>
+        /// <param name="value">The <see cref="Nullable{T}"/> of <see cref="short"/> value to write.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>A <see cref="Task"/> that represents the asynchronous operation.</returns>
+        /// <remarks>The default behaviour is to execute synchronously, returning an already-completed task. Derived
+        /// 
```

Comment:
Move 1551-1562 into shared method? Return bool for whether convertable and the convertedValue as an out